openmdao/core/group.py

```diff
@@ -1661,21 +1661,27 @@ def _setup_connections(self):
             in_units = allprocs_abs2meta_in[abs_in]['units']
 
             if out_units:
-                if not in_units:
+                out_unit_meta = _find_unit(out_units) if out_units else None
+                out_unit_base_powers = any(i >= 1 for i in out_unit_meta._powers)
```

Comment:
since this logic relies on underlying implementation of units, I think it might be better to implement a `is_unitless(units)` function in the units module that does this.. it would also eliminate the code duplication

Reply:
Fixed